Dockerfile

```diff
@@ -0,0 +1,26 @@
+ARG NEXUS_VERSION=3.8.0
+
+FROM maven:3-jdk-8-alpine AS build
+
+ARG PROJECT_NAME=nexus-blobstore-s3
+
+COPY . /${PROJECT_NAME}/
+RUN cd /${PROJECT_NAME}/; mvn package -DskipTests=true;
```

Comment:
QUESTION: Why skip tests? I'd assume you would want this to fail if tests fail?

Reply:
My first idea was trying to gain some building time. In this case, it does not make that much sense considering the number of tests.

As I said, I think we are mixing different concerns here: _library_ building should NOT happen in **Docker** image building. They should be separate and dependent processes: first tests and library building, then *Docker* image building. Is that possible to invoke `docker` from **Maven** build? **Gradle** has such a plugin. I do NOT know **Maven** very well.

I did NOT understand the reasons why you appreciate such a scenario but you must have yours.

Reply:
Generally it allows someone who doesn't have Maven installed to build the project fairly easily. I think that most people coming to these plugins have some amount of Java experience, but having watched the consumers of our R plugin for example, we get just a fair amount of people who want something that does X. Allowing it to be built in Docker makes it a lil more language agnostic, and uses a fairly popular tool in the DevOps toolchain.

If you don't want to make the change that's fine, twas only a suggestion :)